maintainers/README.md

```diff
@@ -1,10 +1,63 @@
 # Nixpkgs Maintainers
 
-The *Nixpkgs maintainers* are people who have assigned themselves to
-maintain specific individual packages. We encourage people who care
-about a package to assign themselves as a maintainer. When a pull
-request is made against a package, OfBorg will notify the appropriate
-maintainer(s).
+Unlike other packaging ecosystems, the maintainer doesn't have exclusive
+control over the packages and modules they maintain. This more fluid approach
+is one reason why we scale to so many packages.
+
+## Definition and role of the maintainer
+
+The main responsibility of a maintainer is to keep the packages they maintain
+in a functioning state, and keep up with updates. In order to do that, they
+are empowered to make decisions over the packages they maintain.
+
+That being said, the maintainer is not alone proposing changes to the
+packages. Anybody (both bots and humans) can send PRs to bump or tweak the
+package.
+
+We also allow other non-maintainer committers to merge changes to the package,
+provided enough time and priority has been given to the maintainer.
+
+For most packages, we expect committers to wait at least a week before merging
+changes not endorsed by a package maintainer (which may be themselves). This should leave enough time
+for the maintainers to provide feedback.
```

Comment:
That is not true for all subsystems. For example the python subsystem is regularly ignoring this on purpose to keep a healthy ecosystem. I am not sure how this alignes with this now.

Reply:
Can you please be clearer about what you mean by this? e.g. Python committers merging bump < 1week because we want to keep up-to-date packages? I do not see why a 1-week wait could not be honored. 

It would just create a 1-week lag, which is not a big deal IMHO.

Reply:
> ignoring this on purpose to keep a healthy ecosystem

Can this be described as «python team considers selves as comaintainers of all the packages inside `python-packages.nix`»?

> It would just create a 1-week lag, which is not a big deal IMHO.

Adding a 1-week lag to a process that works without it needs a good justification why the process will start working better afterwards…

Reply:
> Adding a 1-week lag to a process that works without it needs a good justification why the process will start working better afterwards…

Well, it all boils down again to the position of the package in the dependency tree, some folks do not take this into account and just go on a happy merge spree, some cares about, etc.

Either case, @mweinelt has probably good opinions on that and I trust their judgement on those questions.

Reply:
BTW,

> Can this be described as «python team considers selves as comaintainers of all the packages inside python-packages.nix»?

Yes, it would be good to ask the opinion of the Python team before writing something they may not agree about. :)

cc @FRidh @jonringer @tjni

Reply:
The python ecosystem only ever carries one* specific version for each package, so we often have to touch many packages, to keep everything working. I would guess that only 10% of all 6000+ python packages have active maintainers, so bulk updates have become a requirement for a healthy package set. Ultimately this situation led to a few very active maintainers essentially co-maintaining most of the package set.


<sup>* yes, there are exceptions, but they're exactly that<sup>

Reply:
We definitely don't optimize for giving maintainers a chance to respond, because the definition of maintainer for most Python packages has been: "I added the package, I might be interested in keeping it up to date, and I might be able to help answer some questions about it."

Reply:
Either case, the mentioned sentence is "for most packages", I would argue: we should get this in and refine it for Python ecosystems and similar ecosystems.

Reply:
Let's keep this in mind for after the PR is merged. We need a better mechanism for maintainers to communicate those expectations.